### PR TITLE
Expand use of FlipCoin for unfair choices

### PR DESCRIPTION
**Pull request by ephphatha**
Opening as a draft till I finish going through the non-50/50 rolls.

### CODE REVIEW

File: Source/levels/themes.cpp [hunk 1]
```diff
@@ -577,14 +574,26 @@ void Theme_MonstPit(int t)
 	PlaceThemeMonsts(t, monstrnd[leveltype - 1]);
 }
 
+namespace {
+void SpawnObjectOrSkeleton(unsigned frequency, _object_id objectType, Point tile)
```
Comment (AJenbo) [inline Source/levels/themes.cpp:578]: Looks like `objectType` is alwasy `OBJ_BANNERL`

**Comment by AJenbo** [comment]
Nice with all the improved variable names :)

File: Source/levels/themes.cpp (hunk 1 above)
Comment (ephphatha) [reply Source/levels/themes.cpp:578]: Shouldn't be, it's left right middle middle right left

Comment (AJenbo) [reply Source/levels/themes.cpp:578]: Are you trying to use the Konami Code on me :eyes:

**Review by AJenbo — DISMISSED** [review]

**Comment by AJenbo** [comment]
There are still some cases like in GenerateRoom where this could be applied. I'll let you decide if you want to merge it as is or add more to it:
https://github.com/diasurgical/devilutionX/blob/master/Source/levels/drlg_l1.cpp#L898 (also in l4)

**Comment by AJenbo** [comment]
I'm just going to merge this before we get any more conflicts :D You can always make a follow up for the remaining cases :)

**Comment by glebm** [comment]
Getting this warning now:

```
Source/levels/themes.cpp:190:31: warning: comparison of integer expressions of different signedness: ‘const unsigned int’ and ‘int’ [-Wsign-compare]
  190 |                 if (frequency != -1 && FlipCoin(frequency)) {
      |                     ~~~~~~~~~~^~~~~
```

**Pull request merged**